### PR TITLE
add grad accum for ppo + envpool + impala atari wrapper

### PR DESCRIPTION
**Pull request by 51616**
Use `MultiSteps` wrapper from `optax` to do gradient accumulation to avoid OOM error.

~Note: I'm not sure how to log the learning rate when using the wrapper. I looked through the `agent_state.opt_state` but didn't find the current learning rate. So I commented it out for now.~
Now the code logs the learning rate correctly.

~Here's the result using two random seeds. I'll run one more run tonight.~
Here's the result with 3 random seeds.

image

### CODE REVIEW

File: cleanba/cleanba_ppo_envpool_impala_atari_wrapper_grad_accum.py [hunk 1]
```diff
@@ -0,0 +1,872 @@
...
+        writer.add_scalar("stats/rollout_queue_size", rollout_queue.qsize(), global_step)
+        writer.add_scalar("stats/params_queue_size", params_queue.qsize(), global_step)
+        print(
+            global_step,
+            f"actor_policy_version={actor_policy_version}, actor_update={update}, learner_policy_version={learner_policy_version}, training time: {time.time() - training_time_start}s",
+        )
+
+        # TRY NOT TO MODIFY: record rewards for plotting purposes
+        # Not sure how to log the learning rate when using the MultiSteps wrapper for grad accum
+        # writer.add_scalar("charts/learning_rate", agent_state.opt_state[1].hyperparams["learning_rate"][0].item(), global_step)
```
[the remark this answers is not in the document]
Comment (51616) [reply cleanba/cleanba_ppo_envpool_impala_atari_wrapper_grad_accum.py]: Thank you! Now it should log the current learning rate correctly.

**Comment by vwxyzjn** [comment]
Awesome! Thanks for the PR @51616
